cpu/gd32v/periph/wdt.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2019 Inria
+ * Copyright (C) 2020 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_gd32v
+ * @ingroup     drivers_periph_wdt
+ *
+ * @brief
+ *
+ * @{
+ *
+ * @file        wdt.c
+ * @brief       Independent Watchdog timer for gd32v platforms
+ *
+ * @author      Francisco Molina <francois-xavier.molina@inria.fr>
+ * @author      Koen Zandberg <koen@bergzand.net>
+ */
+
+#include <stdlib.h>
+#include <assert.h>
+#include "macros/units.h"
+
+#include "bitarithm.h"
+#include "cpu.h"
+#include "timex.h"
+
+#include "periph_cpu.h"
+#include "periph/wdt.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define CLOCK_LSI                   (KHZ(40))
+
+#define MAX_RELOAD                (4096U)
+#define MAX_PRESCALER             (6U)
+#define FWDGT_STEP_MS              ((4U * US_PER_MS * MAX_RELOAD) / CLOCK_LSI)
+
+#define FWDGT_CTL_KEY_RELOAD        ((uint16_t)0xAAAA)
+#define FWDGT_CTL_KEY_ENABLE        ((uint16_t)0xCCCC)
+
+#define FWDGT_UNLOCK               ((uint16_t)0x5555)
+#define FWDGT_LOCK                 ((uint16_t)0x0000)
+
+#if ENABLE_DEBUG
+/* wdt_time (us) = LSI(us) x 4 x 2^PRE x RELOAD */
+static inline uint32_t _wdt_time(uint8_t pre, uint16_t rel)
```

Comment:
Maybe add `__attribute__((used))__` instead of using the preprocessor?

Reply:
Sounds good, added that and removed the `#if`